rootfs-builder/rootfs.sh

```diff
@@ -247,6 +245,39 @@ error_handler()
 	fi
 }
 
+detect_go_version()
+{
+	typeset -r yq=$(command -v yq || command -v ${GOPATH}/bin/yq)
+	[ -z "$yq" ] && die "'yq' application not found (needed to parsing minimum Go version required)"
+
+	typeset -r runtimeVersion="${AGENT_VERSION:-master}"
+	typeset -r runtimeVersionsURL="https://raw.githubusercontent.com/kata-containers/runtime/${runtimeVersion}/versions.yaml"
+
+	GO_VERSION=$(curl -sSL "${runtimeVersionsURL}" | $yq r - languages.golang.version)
+	info "Required Go version: $GO_VERSION"
+}
+
+
+compare_versions()
+{
+	typeset -i -a v1=($(awk 'BEGIN {FS = "."} {print $1" "$2" "$3}' <<< $1))
+	typeset -i -a v2=($(awk 'BEGIN {FS = "."} {print $1" "$2" "$3}' <<< $2))
```

Comment:
$3 is not really needed, also would be good to check the len of `v1` and `v2` is at least 2